plugins/tours/tests/phpunit/unit/admin/addDescriptionBeneathPostTitle.php

```diff
@@ -24,58 +24,44 @@
  */
 class Tests_AddDescriptionBeneathPostTitle extends Test_Case {
 
-	/**
-	 * Prepares the test environment before each test.
-	 */
 	protected function setUp() {
 		parent::setUp();
 
 		require_once TOURS_ROOT_DIR . '/src/admin/edit-form-advanced.php';
 	}
 
 	/**
-	 * Test add_description_beneath_post_title() should npt contain view when the post type is 'post'.
+	 * @dataProvider addTestData
 	 */
-	public function test_should_not_contain_view_when_post_type_is_post() {
+	public function testShouldRenderViewAsExpected( $post_type, $expected_html ) {
 		$post = Mockery::mock( 'WP_Post' );
 		Functions\expect( 'get_post_type' )
 			->once()
 			->with( $post )
-			->andReturn( 'post' );
-
-		$view_html = <<<VIEW
-<span class="description">Enter the theme name above for this Cornerstone tour. In the editor below, add each of the venues and locations (city, state) where Cornerstone performed on this tour. Below the editor, enter additional tour information in the box labeled "Past Tour Custom Fields".</span>
-VIEW;
+			->andReturn( $post_type );
 
 		// Fire the rendering function and grab the HTML out of the buffer.
 		ob_start();
 		add_description_beneath_post_title( $post );
 		$actual_html = ob_get_clean();
 
-		$this->assertEmpty( $actual_html );
-		$this->assertNotContains( $view_html, $actual_html );
+		// Test the HTML.
+		$this->assertSame( $expected_html, $actual_html );
 	}
 
-	/**
-	 * Test add_description_beneath_post_title() should contain view when the post type is 'tours'.
-	 */
-	public function test_should_contain_view_when_post_type_is_tours() {
-		$post = Mockery::mock( 'WP_Post' );
-		Functions\expect( 'get_post_type' )
-			->once()
-			->with( $post )
-			->andReturn( 'tours' );
-
-		$view_html = <<<VIEW
+	public function addTestData() {
+		return [
+			'test_should_contain_view_when_post_type_is_tours' => [
```

Comment:
In a data provider, the keys are _not_ passed to the test class. Why have a key for each test data? Context. It can help you to better understand what that specific test data is doing and why.

Question: Do you prefer to have the keys for context? or not? If no, we can delete them. If yes, then go ahead and fix this one plus add them to each of the data provider test data.

Reply:
@hellofromtonya That's what I took away from one of the articles you linked to on data providers; the use of an array key as context for a data set. So yes, I want to keep the contextual array keys. I think they're useful in terms of reading the codebase. I'll make the change for this one data Provider since the intent between datasets is apparent.